iroh-p2p/benches/lru_cache.rs

```diff
@@ -0,0 +1,316 @@
+//! Test the LRU cache implementation.
+//!
+//! These are a few simple tests of the operations we do on empty and full caches.  Mostly
+//! how populated the cache is doesn't seem to affect things much.
+//!
+//! # Running the benchmarks
+//!
+//! Install `cargo-criterion`:
+//!
+//! ```shell
+//! cargo install cargo-criterion
+//! ```
+//!
+//! Run the benchmarks:
+//!
+//! ```shell
+//! cargo criterion -p iroh-p2p
+//! ```
+//!
+//! # Disabled benches
+//!
+//! The bench functions for the `caches` crate are commented out in order not to require a
+//! dependency on the crate.
```

Comment:
Let's remove these entirely, and reposition this PR as swapping out the `caches` crate for `lru`, but also comes with the upside of having a benchmark